CIPs/0008.md

```diff
@@ -0,0 +1,77 @@
+
+- Date: 2020-05-05
+- Author: @nambrot
+- Status: WIP
+
+## Overview
+
+In CIP3, the Celo protocol has started with a simple off-chain storage mechanism mainly focused on claims an account can make about their off-chain identity and communicate to other participants in the network. Over time, there have been additional use cases which have similar access patterns that prompted the creation of `Metadata` in the first place. I.e. data that does not require the expensive cost of "hard consensus", instead just needs to be authenticated by a single party.
+
+Examples include:
+- Transaction metadata such as comments or encrypted messaging in general
+- Profile information such as name or a profile picture
+- Unidirectional attestations like an audit or badges won in The Great Celo Stake Off
+- Signed statements/claims by a party for relase binaries/docker images/package hashes
+
+## Goals
+
+- Expand storage and indexing capabilities of this off-chain data
+- Encode the notion of decryption keys
+- Lightweight messaging system
+
+
+## Proposed Solution
+
+We propose to add another claim, called the `STORAGE` claim:
+```json
+{
+  "type": "STORAGE",
+  "timestamp": 1588886114907,
+  "address": "https://example.com/celo_storage",
+  "filterdataPaths": ".*"
+}
+```
+
+This http address should be referred to as the root. The expectation here is that data can accessed via `root/${dataPath}`. A root can specify a REGEX with the `filterdataPaths` key to indicate what kind of data it stores. An account can make multiple `STORAGE` claims, both to increase availability, as well as effectively sharding this off-chain data. This CIP does not specify what to do in case of conflicts.
+
+Here are some common ways of modeling applications into this format:
+
+- For transaction metadata, the account can encrypt the metadata and put it in a file with the transaction hash as the metadata. Further CIPs could specify a standard for thee structure of that metadata.
+    - `dataPath`: `/transactions/transactionHash`
+    - file contents: Encrypted, structured transaction metadata (comments, or otherwise)
+- Data in relation to other accounts could be communicated to other accounts with:
+    - `dataPath`: `/accounts/otherAccountAddress`
+    - file contents: Structured data such as:
+        - `sessionKeys`: The session key(s) encrypted with dataEncryptionKey of `otherAccount` on `Accounts.sol`
+        - `attestations`: These could be signed data attestating things such an audit or badges. An attested account could also store the attestations from the attestor here, as a client can just verify the signature of the attestor.
+- Profile information could be stored on this system with:
+    - `dataPath`: `/profile/name` or `/profile/picture`
+    - file contents: the data unencrypted, or encrypted to session keys available to the recipients
+- Releases:
+    - `dataPath`: `/releases/hashOfRelease`
+    - file contents: The signature of the key on the hash of the release
```

Comment:
I would be wary of including these "examples" here unless you intend them to actually be implemented. Also could you clarify which parts of these example paths are variable and which are literal?

Reply:
I agree that this shouldn't ultimately be in any kind of spec, but received feedback that examples here were useful in aiding understanding. Updated to make it clearer which are variables vs. literals